src/packages/core/src/types.ts

```diff
@@ -0,0 +1,32 @@
+import type EthereumFlavor from "@ganache/ethereum";
+import type { AnyFlavor, ServerConfig } from "@ganache/flavor";
+import { ExternalConfig, InternalOptions } from "@ganache/options";
+
+type NamespacedServerConfigOptions = {
+  server: ServerConfig;
+};
+
+export type ProviderOptions<F extends AnyFlavor> =
+  (F["flavor"] extends "ethereum"
+    ? {
+        flavor?: F["flavor"];
+      }
+    : {
+        flavor: F["flavor"];
+      }) &
+    Parameters<F["options"]["provider"]["normalize"]>[0];
```

Comment:
The flavor must provide us with an @ganache/options interface, and so we dig into it's normalization method to grab the options the flavor expects from the user before normalization.